examples/src/python/example/python_distribution/hello/fasthello/setup.py

```diff
@@ -5,16 +5,22 @@
 from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
                         unicode_literals, with_statement)
 
+import os
+
 from setuptools import setup, find_packages
 from distutils.core import Extension
 
 
 c_module = Extension(str('c_greet'), sources=[str('c_greet.c')])
 cpp_module = Extension(str('cpp_greet'), sources=[str('cpp_greet.cpp')])
 
+public_version = '1.0.0'
+local_version = os.getenv('_SETUP_PY_LOCAL_VERSION')
```

Comment:
This places responsibility of this existence on the caller, so we would need to document this or throw a warning if it is not present.

Reply:
It would probably be super easy to just check the version when we make the local wheels product and make sure it has a PEP 440 "local" identifier at the end -- then, we allow the possibility of using setup.py projects generated in a different way if we ever need to, but still avoid the name clashing issue!

Reply:
Also, as was very effectively and thoroughly discussed in #5449, and as I have thought about before, there are many more ways to potentially make this part of the setup.py more ergonomic/natural (including making temporary files that the setup.py could import, for example?). I think the env var is a simple and effective answer right now but I'm 100% ready to go deeper into setuptools as need be.

Reply:
TIL the [egg_info](https://setuptools.readthedocs.io/en/latest/setuptools.html#tagging-and-daily-build-or-snapshot-releases) setup command. Maybe this helps?
```
$ ./pants setup-py --run='egg_info -b+bob sdist bdist bdist_wheel' src/python/pants:pants-packaged
...
$ ls -1 dist/pantsbuild.pants-1.9.0.dev0/dist/
pantsbuild.pants-1.9.0.dev0+bob.linux-x86_64.tar.gz
pantsbuild.pants-1.9.0.dev0+bob-py2-none-any.whl
pantsbuild.pants-1.9.0.dev0+bob.tar.gz
```

Reply:
@jsirois I didn't see that comment until now -- thanks! It would be utterly fantastic to be able to avoid making any changes to the setup.py to support this. Will look into using that now.

Reply:
Of *course* it works -- added in 1fd8db129ed121f2338aae2a0561492d30fdb773 so we can do this without any changes to setup.py at all! I should have probably expected this to exist and looked beforehand, so thanks again for mentioning it!